File: module-1/09_Classes_Encapsulation/student-exercise/java/src/main/java/com/techelevator/Elevator.java
package com.techelevator;

public class Elevator {
	int currentLevel;
	int numberOfFloors;
	boolean doorOpen;

	public Elevator() {
		this(1);
	}
	public Elevator(int numberOfFloors) {
		this.numberOfFloors = numberOfFloors;
		this.currentLevel = 1;
	}

	public int getCurrentFloor() {
		return currentLevel;
	}

	public int getNumberOfFloors() {
		return numberOfFloors;
	}

	public boolean isDoorOpen() {
		return doorOpen;
	}
	
	public void openDoor() {
		doorOpen = true;
	}
	
	public void closeDoor() {
		doorOpen = false;
	}
	
	public void goUp(int desiredFloor) {
		boolean isDoorOpen = isDoorOpen();
		if (desiredFloor <= numberOfFloors && desiredFloor > currentLevel && isDoorOpen != true) {
			currentLevel = desiredFloor;
		}
	}
	
	public void goDown(int desiredFloor) {
		boolean isDoorOpen = isDoorOpen();
		if (desiredFloor >= 1 && desiredFloor < currentLevel  && isDoorOpen != true) {
			currentLevel = desiredFloor;
		}
	}
}

File: module-1/09_Classes_Encapsulation/exercise-final/dotnet/Individual.Exercises/Classes/Elevator.cs
namespace Individual.Exercises.Classes
{
    public class Elevator
    {
        /// <summary>
        /// Current elevator level
        /// </summary>
        public int CurrentLevel { get; private set; }


        /// <summary>
        /// Number of levels available for elevator.
        /// </summary>
        public int NumberOfLevels { get; private set; }


        /// <summary>
        /// Is Elevator Door Open?
        /// </summary>
        public bool DoorIsOpen { get; private set; }


        /// <summary>
        /// Creates a new elevator
        /// </summary>
        /// <param name="totalNumberOfFloors">Number of floors in the elevator</param>
        public Elevator(int totalNumberOfFloors)
        {
            this.NumberOfLevels = totalNumberOfFloors;
            this.CurrentLevel = 1;
        }

        /// <summary>
        /// Opens the elevator door
        /// </summary>
        public void OpenDoor()
        {
            DoorIsOpen = true;
        }

        /// <summary>
        /// Closes the elevator door
        /// </summary>
        public void CloseDoor()
        {
            DoorIsOpen = false;
        }

        /// <summary>
        /// Moves the elevator up, as long as the door is closed and the desired floor is not past the last floor
        /// </summary>
        /// <param name="desiredFloor">Desired floor to go to</param>
        public void GoUp(int desiredFloor)
        {
            if (!DoorIsOpen && desiredFloor > CurrentLevel && desiredFloor <= NumberOfLevels)
            {
                CurrentLevel = desiredFloor;
            }
        }

        /// <summary>
        /// Moves the elevator down, as long as the door is closed and desired floor is lower than current but not past 1
        /// </summary>
        /// <param name="desiredFloor">Floor to go to</param>
        public void GoDown(int desiredFloor)
        {
            if (!DoorIsOpen && desiredFloor < CurrentLevel && desiredFloor > 0)
            {
                CurrentLevel = desiredFloor;
            }
        }
    }
}
